tfhe-zk-pok/src/proofs/pke_v2.rs

```diff
@@ -761,7 +783,7 @@ fn prove_impl<G: Curve>(
         )
         .collect::<Box<[_]>>();
 
-    let v = four_squares(sqr(B as u128) - e_sqr_norm).map(|v| v as i64);
+    let v = four_squares(B_squared - e_sqr_norm).map(|v| v as i64);
```

Comment:
the v here should be smaller than (B_squared - e_sqr_norm), which itself is smaller than i64::MAX because we checked above that B_squared is greater than e_sqr_norm.

Reply:
we have a test case that covers this one ?

Reply:
what do you mean ?

Reply:
B_squared is greater than e_sqr_norm if we are inside the noise bounds and smaller if we are outside, so both cases are covered by noise tests

Reply:
that was my question, then all good !